src/vs/code/node/cli.ts

```diff
@@ -319,23 +319,89 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = args['status'] || args['verbose'] || hasReadStdinArg;
+			const envVars: string[] = [];
+			for (const envKey in env) {
+				if (envKey === '_') {
+					continue;
 				}
-			});
+				const value = env[envKey];
+				envVars.push('--env');
+				envVars.push(envKey + '=' + value ?? '');
+			}
+			const openArgs: string[] = ['-n'];
+
+			let tmpfile = '';
+			let tmpfileError = '';
+			if (requiresWait) {
+				const time = new Date().getTime();
+				tmpfile = `${tmpdir()}/vscode-wait-transfer-${time}.log`;
+				tmpfileError = `${tmpdir()}/vscode-wait-transfer-error-${time}.log`;
+				writeFileSync(tmpfile, '');
+				openArgs.push('-W');
+				openArgs.push('--stdout', tmpfile);
+				openArgs.push('--stderr', tmpfileError);
+				argv.push('--wait');
+			}
+			const argsArr: string[] = [];
+			const execPathToUse = args['exec-path'] ?? process.execPath;
+			argsArr.push('-a', execPathToUse);
+			argsArr.push(...envVars, ...openArgs, '--args', ...argv.slice(2));
+			child = spawn('open', argsArr, options);
+
+			if (requiresWait) {
+				const openPromise = (child: ChildProcess) => new Promise<void>((c) => {
+					console.log('Temp file location: ' + tmpfile);
+					if (args['verbose']) {
+						const stream = openSync(tmpfile, 'r');
+						const bufferSize = 50;
+						const buffer = Buffer.alloc(bufferSize);
+						const interval = setInterval(() => {
+							const readAmount = readSync(stream, buffer, 0, bufferSize, null);
+							process.stdout.write(buffer.toString(undefined, 0, readAmount));
+						}, 50);
+						child.on('exit', () => {
+							setTimeout(() => {
+								clearTimeout(interval);
+								closeSync(stream);
+								c();
+							}, 1500);
+						});
+					} else if (args['status']) {
+						child.on('exit', () => {
```

Comment:
I don't really see why `status` couldn't use the same solution as `verbose` and just poll?

Reply:
I initially set it up this way so that it doesn't need to poll, and also so that the output has trimmed lines.
The polling solution works, but there are extra newlines at the beginning and end, and there is also a `setTimeout` after the child process closes to capture any remaining output, which adds a delay.
I think if there is a better way to get the data rather than polling, the perf problem can be fixed.